fuzz/fuzz_targets/preserve_rdata.rs

```diff
@@ -158,22 +182,30 @@ fn split_rrs(buffer: &[u8], query_count: u16, rr_count: u16) -> Vec<Record<'_>>
         output.push(Record { r#type, rdata });
     }
 
-    output
+    Ok(output)
 }
 
 const LABEL_TYPE_MASK: u8 = 0b1100_0000;
 const COMPRESSED_LABEL_TYPE: u8 = 0b1100_0000;
 
 /// Determines the encoded length of a name inside a DNS message.
-fn name_length(input: &[u8]) -> usize {
+fn name_length(input: &[u8]) -> Result<usize> {
     let mut offset = 0;
 
-    while input[offset] != 0 && input[offset] & LABEL_TYPE_MASK != COMPRESSED_LABEL_TYPE {
-        let length = input[offset];
-        offset += 1 + length as usize;
+    loop {
```

Comment:
How about rewriting this in terms of `for (i, b) in foo.iter().enumerate()`?

Reply:
I don't think that makes sense here, because we jump ahead by several characters on each iteration. We're just visiting the first byte delimiting each label, not each byte of the name.